Rom key documentation update

pamaury opened the pull request:
Document how to generate keys and also document how the key slots work and what they are.

pamaury commented:
@cfrantz Since the key slots are earlgrey-specific, I am thinking that it makes sense to put in the earlgrey documentation but I don't have a high-level understanding of the documentation structure to make a good decision.

Also I don't know if it makes sense for the source of truth to be a hjson file, like for other things such as memory map or pinout, and for this table to be auto-generated as part of the documentation?

cfrantz commented on hw/top_earlgrey/doc/design/index.md:
```diff
@@ -178,6 +178,17 @@ It is also available for code storage, though that is not its intended purpose.
 
 The base address of the ROM, Flash, and SRAM are given in the address map section later in this document.
 
+### Secure boot
+
+Earlgrey follows the [Secure Boot](security/specs/secure_boot/index.md) specification. The `ROM` has 3 key slots
+which are allocated as follows:
+
+| Slot | Role | Name |
+|------|------|------|
+| 0 | test | `test 0` |
+| 1 | dev | `dev 0` |
+| 2 | prod | `prod 0` |
+
```
@alphan @moidx What should we specify as the full keyset for Earlgrey?

My opinion: 3-2-1 for PROD/DEV/RMA keys.

moidx replied:
SGTM

* PROD: 3
* DEV: 2
* TEST/RMA: 1

This means we have to be very careful about how we manage the `TEST/RMA` key as there is no back up for that configuration.

cfrantz replied:
I'm not opposed to 3-2-2 (add an extra TEST/RMA key).  I think we have the space for it (both in ROM and in OTP for an enable/disable byte).  I thought I'd start with 3-2-1 and see what other people thought about it.

cfrantz commented on sw/device/silicon_creator/rom/keys/README.md:
````diff
@@ -99,3 +144,5 @@ c55ad015
 9f86d081
 ```
+
+[ott]: #
````
What is meant to go here?

pamaury replied:
Sorry, that was meant to be a link to the doc of opentitantool but I could not find it (does it exist ?). Probably should remove that.

pamaury commented:
I added the table in #17364 since it seems to make sense to do it at the same time. I have split this PR between the generic doc and the earlgrey specific one in #17474

cfrantz commented on doc/security/specs/secure_boot/README.md:
```diff
@@ -77,13 +77,31 @@ Devices produced before the key is known to be compromised are not protected by
+
+| Key Type | LC_TEST | LC_RMA | LC_DEV | LC_PROD |
+|----------|---------|--------|--------|---------|
+| `dev`    | X       | X      |        |         |
+| `test`   | X       | X      | X      |         |
```
It looks like you have the key-types and valid LC states for `dev` and `test` keys mixed up: The `dev` key should be valid for test/rma/dev while the `test` key should be valid only for test/rma.

pamaury replied:
Indeed, I swapped the two, thanks for catching this.

moidx commented on doc/security/specs/secure_boot/README.md:
```diff
@@ -77,13 +77,31 @@ Devices produced before the key is known to be compromised are not protected by
-* test (manufacturing and testing, only for devices in the `TEST_UNLOCK` lifecycle state)
-* prod (production, only for devices in the `PROD` or `PROD_END` lifecycle states)
+* `dev`: development, only for devices in the `DEV` lifecycle state
+* `test`: manufacturing and testing, only for devices in the `TEST_UNLOCK` lifecycle state
```
nit: Only for devices in `TEST_UNLOCKED` and `RMA` lifecycle states.

pamaury replied:
I have just updated the text and table, there was an error because it did not reflect the current code (which only allows `dev` keys in `DEV`). I believe #17364 was about allowing `dev` keys in `TEST` and `RMA`?

moidx commented on doc/security/specs/secure_boot/README.md:
```diff
@@ -77,13 +77,31 @@ Devices produced before the key is known to be compromised are not protected by
 
+The `ROM` has `N` key slots (the exact number depends on the `ROM`) numbered from `0` to `N-1`.
+The `CREATOR_SW_CFG_SIGVERIFY_RSA_KEY_EN` item in the [OTP](otp-mmap) can be used to invalidate a key at
+production time. This item consists of several little-endian 32-bit words. Each word contains four 8-bit hardened booleans
```
s/production time/manufacturing time/ to disambiguate. The production term is sometimes used to refer to in-field deployed devices.

moidx commented on sw/device/silicon_creator/rom/keys/README.md:
````diff
@@ -1,8 +1,53 @@
+not run the tool in the current directory.
+
+- Exporting a key to a header for use in the ROM:
+```
````
nit: Here and in other code blocks, can you use ```console to render inline comments with a different color?

moidx commented on sw/device/silicon_creator/rom/keys/README.md:
````diff
@@ -1,8 +1,53 @@
+$ bazel run //quality:clang_format_fix
+```
+
+**Remark**: it is recommended to specify the licence at the beginning of the file
````
nit: Since presubmit fail you can just say that, i.e. don't forget to add the copyright comment at the beginning of the file.

HU90m commented on sw/device/silicon_creator/rom/keys/README.md:
````diff
@@ -1,11 +1,55 @@
 ```
+
+- Generating a key pair in the current directory:
+```console
+$ bazel run //sw/host/opentitantool -- rsa key generate <outdir> <basename>
````
```suggestion
```sh
bazel run //sw/host/opentitantool -- rsa key generate $outdir $basename
```

* I think removing `$` would allow for easier copy and pasting for users.
* Also if `basename` is a variable, people can set the variable and are then able to copy and paste all the commands. `sh` highlights variables nicely too.

This is just an opinion though, so there is no need to block this PR. I'll open a PR adding to the style guide to see if there is any consensus on this stuff.

engdoreis reviewed (approved):
Thanks @pamaury for the PR. It LGTM.
